CDU management profile and specification

mraineri opened the pull request:
Proposal for Redfish requirements for a CDU.

Still need to provide a Markdown file for a specification that describes this in text.

mick-jones-vertiv commented:
I expect a new baseline for Cooling Equipment to exist at some point which will be inherited by this. Refer to column D in this: https://docs.google.com/spreadsheets/d/1XP0wLm1oMLiDfpaf5FXuIATc3lkqokq9Gr05OYNDSEE/edit#gid=0

mraineri commented:
> I expect a new baseline for Cooling Equipment to exist at some point which will be inherited by this. Refer to column D in this: https://docs.google.com/spreadsheets/d/1XP0wLm1oMLiDfpaf5FXuIATc3lkqokq9Gr05OYNDSEE/edit#gid=0

Do you specifically mean that you want to break down the profiles a bit to maximize reuse? So, have one common cooling baseline, and then have the CDU profile inherit from that?

mraineri commented:
For "Infra Baseline", would it be restricted to "cooling infra"? I think supporting power infra in a common infra baseline would be difficult.

mick-jones-vertiv commented:
Yeah that’s the scope. I misspoke and it should be ‘cooling infra’.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Mike Raineri ***@***.***>
Sent: Saturday, March 23, 2024 12:15:11 AM
To: opencomputeproject/HWMgmt-OCP-Profiles ***@***.***>
Cc: Jones, Mick Bruce ***@***.***>; Comment ***@***.***>
Subject: Re: [opencomputeproject/HWMgmt-OCP-Profiles] CDU management profile and specification (PR #67)


CAUTION: This email originated from outside of Vertiv. Do not click links or open attachments unless you recognize the sender and know the content is safe.


For "Infra Baseline", would it be restricted to "cooling infra"? I think supporting power infra in a common infra baseline would be difficult.

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https://github.com/opencomputeproject/HWMgmt-OCP-Profiles/pull/67*issuecomment-2015078880__;Iw!!C8PT2x6w!ur_53Shaps0XxqjeKRnHkJNcdAaOJoE_xOT5MWTEWM5YKsRZ6KkRfppo8SA64V90cyY4ZkMYU7EEfNc37F_OnSJ_i9IZ5w$>, or unsubscribe<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/ATFDC52OQGX2OTPWWAFTAQ3YZQVF7AVCNFSM6AAAAABDEYNCUOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMJVGA3TQOBYGA__;!!C8PT2x6w!ur_53Shaps0XxqjeKRnHkJNcdAaOJoE_xOT5MWTEWM5YKsRZ6KkRfppo8SA64V90cyY4ZkMYU7EEfNc37F_OnSL8Dm1E9g$>.
You are receiving this because you commented.Message ID: ***@***.***>

CONFIDENTIALITY NOTICE: This e-mail and any files transmitted with it are intended solely for the use of the individual or entity to whom they are addressed and may contain confidential and privileged information protected by law. If you received this e-mail in error, any review, use, dissemination, distribution, or copying of the e-mail is strictly prohibited. Please notify the sender immediately by return e-mail and delete all copies from your system.

mick-jones-vertiv commented:
Hi Mike, I did a PR against your fork/branch which contains changes from review with OCP HMLC community for the CDU profile. We can run this revised profile against our revised public-cooling-unit mockup cleanly.

If it makes it easier I have attached the changed file here as it may be easier to do a diff locally and push to your fork instead of trying to resolve changes from my PR.

[OCPCoolantDistributionUnitManagement.v0_9_0.json](https://github.com/opencomputeproject/HWMgmt-OCP-Profiles/files/14830184/OCPCoolantDistributionUnitManagement.v0_9_0.json)

mick-jones-vertiv commented:
In `LiquidCooling\OCPLiquidCoolingBaseline.v0_9_0.json` the substitution variable `{CoolingUnitType}` does not work in Interop Validator because the regex is looking for an entry ending with "Id". I've fixed this by replacing all occurrences of `CoolingUnitType` with `CoolingUnitTypeId` in this profile.

mraineri commented:
> In `LiquidCooling\OCPLiquidCoolingBaseline.v0_9_0.json` the substitution variable `{CoolingUnitType}` does not work in Interop Validator because the regex is looking for an entry ending with "Id". I've fixed this by replacing all occurrences of `CoolingUnitType` with `CoolingUnitTypeId` in this profile.

Got it; changed it. We might want to see if this is something we should push into the interop validator to better handle.

mraineri commented on LiquidCooling/OCPCoolantDistributionUnit.v1_0_0.json:
```diff
@@ -0,0 +1,208 @@
+                }
+            }
+        },
+        "Filter": {
```
Will need to make this "IfImplemented"; speaking with some CDU folks, there are some designs that simply don't have filters.

mraineri replied:
Made change for this; leaving open as a remind for others to review to ensure agreement.

mick-jones-vertiv replied:
Agreed
